### ISSUE TITLE
Expected ")" but found "22" (when running any new project)

### ISSUE THREAD

**Issue by mohanadkandil** [withastro/astro#5518]
### What version of `astro` are you using?

latest

### Are you using an SSR adapter? If so, which one?

no

### What package manager are you using?

npm

### What operating system are you using?

windows 10

### Describe the Bug

Whenever install just a simple new project using NPM or PNPM, I got this when I run `pnpm/npm run dev`
```
Expected ")" but found "22"
  File:
    D:/x/x/x/x/src/pages/index.astro:23:142
  Stacktrace:
```

### Link to Minimal Reproducible Example

n/a

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by marcfielding1** [comment]
This doesn't provide any information anyone can help you with, can you show your index.astro please, or is it the default?

**Comment by Princesseuh** [comment]
This happened because of a `'` inside the project's file path

**Comment by wulinsheng123** [comment]
> This happened because of a `'` inside the project's file path, the user's folder was named `Project '22/x/thing`

![image](https://user-images.githubusercontent.com/27261522/205489383-c0963ead-fde8-4aa4-ade0-8c6532c5e10f.png)

Why doesn't it happen on my side?

**Comment by Princesseuh** [comment]
> > This happened because of a `'` inside the project's file path, the user's folder was named `Project '22/x/thing`
> 
> ![image](https://user-images.githubusercontent.com/27261522/205489383-c0963ead-fde8-4aa4-ade0-8c6532c5e10f.png)
> 
> Why doesn't it happen on my side? 

Might only be happening on Windows perhaps?

**Comment by HiDeoo** [comment]
I encountered the same issue on macOS with an Astro project created in a directory containing a single quote in its name, e.g. `name'23`.

After some investigations, this issue seems to be resolved in `astro@2.0.0-beta.2` which upgrades the `@astrojs/compiler` to `v0.32.0`.

The issue was related to the output code by the compiler being similar to:

```ts
const $$Astro = $$createAstro(
	"/Users/hideo/Temp/astro/test'23/src/pages/index.astro",
	'',
	'file:///Users/hideo/Temp/astro/test'23/'
	//                                  ^ extra unescaped single quote
);
```

The last `projectRoot` path argument of the `$$createAstro()` call was containing the unescaped single quote of the directory name in a single quoted string, which is invalid code.

The Astro version `2.0.0-beta.2` introduces changes from @bluwy to both [the compiler](https://github.com/withastro/compiler/pull/693) and [astro](https://github.com/withastro/astro/pull/5803) that removes the `site` and `projectRoot` options from `$$createAstro()` in favour of the `astroGlobalArgs` option, meaning the output code from the compiler is now similar to this:

```ts
const $$Astro = $$createAstro('https://astro.build');
```

This prevents the issue from happening.

**Comment by Princesseuh** [comment]
@HiDeoo Thank you for investigating this! Even though it wasn't intended, it is wonderful that this got fixed. 

I'll close this issue now, thank you everyone for reporting, helping and investigating this.
